src/boot/efi/util.c

```diff
@@ -769,6 +770,107 @@ EFI_STATUS make_file_device_path(EFI_HANDLE device, const char16_t *file, EFI_DE
         return EFI_SUCCESS;
 }
 
+#define QEMU_KERNEL_LOADER_FS_MEDIA_GUID \
+        &(const EFI_GUID) { 0x1428f772, 0xb64a, 0x441e, {0xb8, 0xc3, 0x9e, 0xbd, 0xd7, 0xf8, 0x93, 0xc7 }}
+
+#define QEMU_BOOT_ORDER_GUID \
+        &(const EFI_GUID) { 0x668f4529, 0x63d0, 0x4bb5, {0xb6, 0x5d, 0x6f, 0xbb, 0x9d, 0x36, 0xa4, 0x4a }}
+
+/* detect direct boot */
+bool is_qemu_direct_boot(EFI_HANDLE device) {
```

Comment:
(It's not strictly qemu only)
```suggestion
bool is_direct_boot(EFI_HANDLE device) {
```